src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs

```diff
@@ -49,13 +46,22 @@ public void ConfigureServices(IServiceCollection services)
             services.AddDbContext<DdsInstrumentationContext>(options => options
                 .UseNpgsql(Configuration.GetConnectionString("DdsInstrumentation"))
                 .UseSnakeCaseNamingConvention());
-            
+
+            var ddsConnectionString = Configuration.GetConnectionString("Dds");
             services.AddDbContext<DdsContext>(options => options
-                .UseNpgsql(Configuration.GetConnectionString("Dds"))
+                .UseNpgsql(ddsConnectionString)
                 .UseSnakeCaseNamingConvention());
 
             services.AddMemoryCache();
-            services.AddDistributedMemoryCache();
+            services.AddDistributedPostgreSqlCache(setup =>
+            {
+                setup.ConnectionString = ddsConnectionString;
+                setup.SchemaName = "public";
+                setup.TableName = "__dist_cache";
+                setup.CreateInfrastructure = !WebHostEnvironment.IsProduction();
+                setup.DefaultSlidingExpiration = TimeSpan.FromMinutes(20); // TODO - is this right?
```

Comment:
I think we're setting this explicitly when we use it - how do we set session expiry?

Reply:
That is - is this the same as options.IdleTimeout? very probably not the same.

Reply:
`DefaultSlidingExpiration` and session `IdleTimeout` differ. I had a look at implementation of session and it sets sliding expiry, whereas the above is used if we don't explicity set an expiry.

From dotpeek:

```cs
await _cache.SetAsync(
     _sessionKey,
     stream.ToArray(),
     new DistributedCacheEntryOptions().SetSlidingExpiration(_idleTimeout),
     cts.Token);
```